builtin/logical/mssql/secret_creds.go

```diff
@@ -118,13 +118,12 @@ func (b *backend) secretCredsRevoke(ctx context.Context, req *logical.Request, d
 	defer rows.Close()
 
 	for rows.Next() {
-		var loginName, dbName, qUsername string
-		var aliasName sql.NullString
+		var loginName, dbName, qUsername, aliasName sql.NullString
 		err = rows.Scan(&loginName, &dbName, &qUsername, &aliasName)
 		if err != nil {
 			return nil, err
 		}
-		revokeStmts = append(revokeStmts, fmt.Sprintf(dropUserSQL, dbName, username, username))
+		revokeStmts = append(revokeStmts, fmt.Sprintf(dropUserSQL, dbName.String, username, username))
```

Comment:
I'm curious about how the change to username from string to sql.NullString behaves here. NullString doesn't implement any of the marshaling interfaces (that I can see) so what happens when it's pulled into `dropUserSQL`, and why is it not using `.String` like dbName?

Reply:
I'm uncomfortable with the (I believe undefined) behaviour of "USE []".  Even if it behaves fine now, it might do something different in the future, which could lead to us dropping the user from the wrong DB.

Moreover the MSSQL docs for DROP USER say "Removes a user from the current database."  So what would we expect our DROP USER to do when our query didn't return any DBs they belong to?

I think you should go with the check/continue approach.

Reply:
@ncabatoff all good points! I will use that approach.